[Storage] Add StaticWebSite Dataplane Support

blueww opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

<!-- Please add a brief description of the changes made in this PR -->


This PR is to add the static Web data plane support.
It already pass review in https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/88

Since the XSCL still not released, this based on private XSCL and DMlib (So build won't pass). It's just for a early code review to reduce the risk for the late code review. 
I will update it when XSCL released.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

blueww commented:
@praries880 
This PR is just for an early review before XSCL is released, to reduce the risk of the review the PR later, since we need to release it after 1 week of XSCL release.

I will definitely update it after XSCL is released.
And I don't see a conflict now. 
Would you please help to review, and see it any change need for this feature?

praries880 commented:
@blueww can you fix the build breaks in both the Jenkin and net core builds

blueww commented:
@praries880 
The build failure is caused by the XSCL /DMlib still not released, so I use the not signed private build.
I send the PR and would like to get an early code review before the XSCL real release, to reduce the late review risk.

Would you please help to review before XSCL release, and see it any change need for this feature?

praries880 commented:
@blueww  will review the code, will mark it as do not merge though :)

praries880 commented on src/ResourceManager/Storage/Commands.Management.Storage/Commands.Management.Storage.csproj:
```diff
@@ -61,8 +61,8 @@
     <Reference Include="Microsoft.Data.Services.Client, Version=5.8.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>..\..\..\packages\Microsoft.Data.Services.Client.5.8.2\lib\net40\Microsoft.Data.Services.Client.dll</HintPath>
     </Reference>
-    <Reference Include="Microsoft.WindowsAzure.Storage, Version=9.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
assuming this will be undone once the new SDK is released.

blueww replied:
Yes, I will update all the reference when the XSCL is release. Currently remove the “PublicKeyToken” only to make the private XSCL work.

praries880 commented on src/Storage/Commands.Storage/Common/Cmdlet/DisableAzureStorageStaticWebsite.cs:
```diff
@@ -0,0 +1,83 @@
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        // Overwrite the useless parameter
```
What are these about? to set them in the tests?

blueww replied:
This parameter will show the user the StaticWebsite setting after the change. So they can have a check.
This is aligned with other serverproperty cmdlets like
 
enable/disable-AzureStorageDeleteRetentionPolicy
Set-AzureStorageServiceLoggingProperty
Set-AzureStorageServiceMetricsProperty
Set-AzureStorageServiceProperty

praries880 replied:
The comment says "Overwrite the useless parameter".. these dont seem useless :)

blueww commented:
@praries880 

I have updated the PR to the released XSCL and private signed DMlib. 
(for preview module sign pricate SDK is allowed)

I have also updated the change log and module version. (not sure if 4.4.1 is correct version)
Your command are all answered.
Would you please help to review?

praries880 commented on src/Storage/Commands.Storage/Common/Cmdlet/EnableAzureStorageStaticWebsite.cs:
```diff
@@ -0,0 +1,88 @@
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        // Overwrite the useless parameter
```
Are these really useless?

blueww replied:
Yes, these are useless. These parameter are for the long running blob/file content transfer cmdlets. But for this cmdlet, it's just a property setting, and can finish very quickly, don't need set special timeout and concurrent.

blueww commented:
On Demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/976/
sign: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/ps-sign/824/
